hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/DefaultS3ClientFactory.java

```diff
@@ -361,6 +366,13 @@ private static URI getS3Endpoint(String endpoint, final Configuration conf) {
    */
   private static Region getS3RegionFromEndpoint(String endpoint) {
 
+    // S3 VPC endpoint parsing
+    Matcher matcher = VPC_ENDPOINT_PATTERN.matcher(endpoint);
+    if(matcher.find()) {
+      LOG.debug("Endpoint {} is vpc endpoint; parsing", endpoint);
+      return Region.of(matcher.group(1));
```

Comment:
add a debug log saying "mapping to vpce"

Reply:
could you log the group(1) value in the debug log so we can see exactly which one it is. region and endpoint bindings are the cause of many of our support calls.